### ISSUE TITLE
Add an override for the number of asyncrunner threads for VC service

### ISSUE THREAD

**Issue by corverroos** [Consensys-Incorporated/teku#6667]
### Updated Description

See [this comment](https://github.com/ConsenSys/teku/issues/6667#issuecomment-1379843164) for a better description of the requested feature

### Original Description

We at Obol are running a "performance/load testing" DVT cluster using teku as the VC. The cluster has 500 active validators on goerli. The cluster is sporadically missing attestations due to "inactivity" from teku, ie, the VC doesn't perform some attestation duties. We created an internal ticket: https://github.com/ObolNetwork/charon/issues/1585

The VC logs the following for those missed slots:
```
WARN  - Skipping aggregation for slot 4694957 due to unexpected delay in slot processing
```

Digging into the teku code, I'm wondering if this due to [Serviceutils#calculateMaxThreads](https://github.com/ConsenSys/teku/blob/master/infrastructure/serviceutils/src/main/java/tech/pegasys/teku/service/serviceutils/ServiceConfig.java#L95)?
 - In a DVT cluster, some blocking API calls take much longer than in traditional single validator setups.
 - This is because we need to perform a bunch of internal p2p network hops for consensus across the whole cluster.
 - If other peers in the cluster are having issues, then the data might never be available and those queries need to timeout.
 - If teku is only using `max(Runtime.getRuntime().availableProcessors(), 5)` concurrent threads, then it seems likely that this is causing the problem.

Would it be possible to increase this number or make it configurable?

### Steps to Reproduce (Bug)

Run a Obol DVT cluster with 4 nodes and 500 active validators on goerli.

### Versions (Add all that apply)
* Software version: 22.12.0
* Java version: 17.0.5
* OS Name & Version: Ubuntu (GKE)
* Docker Version: Unknown (GKE)
* Cloud VM, type, size: GKE node, JAVA_OPTS="-Xmx4g -Xms4g

**Comment by lucassaldanha** [comment]
Hi,

Thanks for your report.

We have nodes running in Goerli with 5,000 validators and we are not seeing this problem. Our nodes have close to 100% inclusion rate, and we have not seen this warning in our logs.

Our node is running on AWS, and has the following configuration:

Instance type: m6a.2xlarge

- 8 vCPUs
- 32 GB RAM
- 12.5 Gbps Network Bandwidth

We are using EBS gp3 storage with the following configuration:

- 3,000 IOPS 

I checked our configuration and we are using:

- -Xmx5g for heap size

One option that we are using that could be relevant in your case, is an experimental config `--Xrest-api-validator-threads` that configure the number of threads being used to handle validator api requests. **By default Teku uses 1**, **but we are using 4**:

- --Xrest-api-validator-threads=4


Let me know how your server compare with our reference Goerli node. Also, if you do have a chance to play around the `--Xrest-api-validator-threads` option I'd love to hear the results.

References:
- https://aws.amazon.com/ec2/instance-types/m6a/
- https://aws.amazon.com/ebs/features/
- https://github.com/ConsenSys/teku/blob/master/teku/src/main/java/tech/pegasys/teku/cli/options/BeaconRestApiOptions.java#L127-L132

**Comment by corverroos** [comment]
Thanks @lucassaldanha , but our use-case is a bit different. We are **not** a running normal/traditional validator like you mentioned, we are running a [Obol DVT cluster](https://docs.obol.tech/docs/int/key-concepts#distributed-validator-node) with multiple Tekus as the VCs, Obol's charon nodes as the DVT middleware, and lighthouse as the beacon nodes. 

The difference between a traditional validator and a distributed validator is the we split the validator key into keyshares across multiple VCs. We then introduce a middleware layer that does consensus on data to sign and does threshold aggregation to recombine partial signature before broadcasting them back up to the beacon node. The middleware layer therefore introduces multiple network hops (latency). See our [docs](https://obol.tech/) for more info.

What that means is that the beacon API calls that the VC does has MUCH more latency than a traditional validator (since it doesn't talk with the BN, but rather with the DVT middleware client). And if there are problem in the DVT cluster, then many beacon api calls that the VC does might never return (since the data is never available), that means the calls block until they timeout. The teku beacon API http timeout is 10s I think (so almost 2 slots). 

Since the VC asyncrunner only uses a few threads (from 5 to #cpus) it is easy for all those threads to be blocked waiting for http timeout. During this time, no other duties can be performed. 

This isn't a problem in your 5000 validator goerli setup since the VC talks directly to the BN (no DVT middleware) and the BN is properly resourced so always returns quickly (with  milliseconds or max a one or two seconds). Beacon api calls should never really block forever waiting for http timeout.

Would it be possible to add an override for the VC number of asyncrunner threads, similar to the beacon API override you mentioned? Something like `--Xvalidator-client-runner-threads=20`

**Comment by lucassaldanha** [comment]
@corverroos thank you for the explanation. I will raise this with our product manager to discuss this feature request.

At the moment the development team is 100% focused on the Capella upgrade, so realistically we won't be implementing any new feature requests until the upgrade (ETA March 2023).

**Comment by benjaminion** [comment]
Hey all. How confident are you that this fix would resolve the issue you are seeing? Is it feasible for Obol to try patching Teku to confirm that the fix is effective? Just want to make sure that it's the right fix before we commit, and it's difficult for us to test on our side.

**Comment by corverroos** [comment]
Yes sure, we can test the fix on our side and let you know if it works 🚀

**Comment by corverroos** [comment]
Closing as fixed by #6761
